nexus/sdk-go.md

```diff
@@ -0,0 +1,352 @@
+
+Temporal is leveraing the Nexus RPC protocol to facilitate calling across namespace and cluster and boundaries.
+
+### Prior Art
+
+- [Nexus over HTTP Spec](https://github.com/nexus-rpc/api/blob/main/SPEC.md)
+- [Nexus SDK Go](https://github.com/nexus-rpc/sdk-go)
+
+### This Proposal
+
+This proposal discusses exposing Nexus concepts in the Temporal Go SDK, reusing definitions from the Nexus Go SDK.
+
+> NOTE: The proposed APIs will all be marked experimental until we get user feedback and we feel they've matured enough.
+
+## Temporal Operations
+
+### NewSyncOperation
+
+`NewSyncOperation` creates a synchronous operation, passing the client the worker was created with as an argument to the
+handler. Sync operations are useful for exposing short-lived Temporal client requests, such as signals, queries, sync
+update, list workflows, etc...
+
+```go
+package temporalnexus
+
+import "github.com/nexus-rpc/sdk-go/nexus"
+
+func NewSyncOperation[I any, O any](
+	name string,
+	handler func(context.Context, client.Client, I, nexus.StartOperationOptions) (O, error),
+) nexus.Operation[I, O]
+```
+
+> NOTE: this is a simple wrapper around the Nexus Go SDK's
+> [NewSyncOperation](https://pkg.go.dev/github.com/nexus-rpc/sdk-go@v0.0.7/nexus#NewSyncOperation), which can be used in
+> sync operations that don't require a Temporal client.
+
+**Usage**:
+
+```go
+import (
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporalnexus"
+)
+
+opGetStatus := temporalnexus.NewSyncOperation("get-status", func(ctx context.Context, c client.Client, id string, opts nexus.StartOperationOptions) (int, error) {
+	res, err := c.QueryWorkflow(ctx, id, "", "some-query", nil)
+	if err != nil {
+		return 0, err
+	}
+	var ret int
+	return ret, res.Get(&ret)
+})
+
+// Operations don't have to return values.
+opContinue := temporalnexus.NewSyncOperation("continue-processing", func(ctx context.Context, c client.Client, id string, opts nexus.StartOperationOptions) (nexus.NoValue, error) {
+	return nil, c.SignalWorkflow(ctx, id, "", "continue-processing", nil)
+})
+```
+
+### NewWorkflowRunOperation
+
+```go
+package temporalnexus
+
+import (
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/sdk/client"
+)
+
+// NewWorkflowRunOperation map an operation to a workflow run.
+func NewWorkflowRunOperation[I, O any](
+	name string,
+	workflow func(internal.Context, I) (O, error),
+	getOptions func(context.Context, I, nexus.StartOperationOptions) (client.StartWorkflowOptions, error),
+) nexus.Operation[I, O]
+
+type WorkflowRunOperationOptions[I, O any] struct {
+	// Operation name.
+	Name string
+	// Workflow function to map this operation to. The operation input maps directly to workflow input.
+	// The workflow name is resolved as it would when using this function in client.ExecuteOperation.
+	// OptionsGetter must be provided when setting this option. Mutually exclusive with Handler.
+	Workflow func(internal.Context, I) (O, error)
```

Comment:
```suggestion
	Workflow func(workflow.Context, I) (O, error)
```

I think we may be able to avoid type aliases in the `temporalnexus` package, but understood if we cannot

Reply:
Looks like it's possible. Thanks!

Reply:
Fixed.